starter-code/src/viking.js

```diff
@@ -1,11 +1,102 @@
 // Soldier
-function Soldier() {}
+function Soldier(healthArg,strengthArg) {
+    this.health=healthArg;
+    this.strength=strengthArg;
 
+}
+
+Soldier.prototype.attack = function (){
+    return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function (damage) {
+    this.health -= damage;
+}
 // Viking
-function Viking() {}
+function Viking(naming,health,strength) {
+    Soldier.call(this,health,strength);
+    this.name=naming;
+}
+
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor=Viking;
+
+Viking.prototype.receiveDamage = function (damage){
+    this.health -= damage;
+    if (this.health >0) { 
+        return this.name +' has received '+ damage +' points of damage';
+    } else {
+        return this.name + ' has died in act of combat';
+    }
+}
+
+Viking.prototype.battleCry = function () {
+    return 'Odin Owns You All!';
+} 
 
 // Saxon
-function Saxon() {}
+function Saxon(health,strength) {
+    Soldier.call(this,health,strength);
+
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor=Saxon;
+
+Saxon.prototype.receiveDamage = function(damage) {
+    this.health -= damage;
+    if (this.health >0) { 
+        return 'A Saxon has received '+ damage +' points of damage';
+    } else {
+        return 'A Saxon has died in combat';
+    }
+}
 
 // War
-function War() {}
+function War() {
+    vikingArmy =[];
+    saxonArmy =[];
+}
+
+War.prototype.addViking = function (vikingSoldier){
+    War.vikingArmy.push(vikingSoldier);
+}
+
+War.prototype.addSaxon = function (saxonSoldier){
+    War.saxonArmy.push(saxonSoldier);
+}
+
+War.prototype.vikingAttack = function (){
+    var saxonSoldierIndex=Math.floor(Math.random()*War.saxonArmy.length);
+    var vikingSoldierIndex=Math.floor(Math.random()*War.vikingArmy.length);
+    var returnMessage ='';
+    returnMessage = this.saxonArmy[saxonSoldierIndex].receiveDamage(this.vikingArmy[vikingSoldierIndex].strength);   
+
+    var deadSaxonSoldierIndex = this.saxonArmy.findIndex(function (saxonSoldier){ return saxonSoldier.health <= 0;})
```

Comment:
You have the index (from line 71), don't calculate it again, only check if health is 0 or less.